ReactiveCocoa/Objective-C/RACSignal.m

```diff
@@ -107,31 +107,36 @@ - (RACSignal *)bind:(RACStreamBindBlock (^)(void))block {
 	return [[RACSignal createSignal:^(id<RACSubscriber> subscriber) {
 		RACStreamBindBlock bindingBlock = block();
 
-		NSMutableArray *signals = [NSMutableArray arrayWithObject:self];
+		__block NSUInteger signalsCount = 1;   // indicates self
```

Comment:
Didn't notice OSAtomicDecrement32Barrier returning the new value. I think it really works! When `signalsCount ` reaches zero, only the last signal holds the subscriber and disposable, so needn't to worry about atomicity.